### PR TITLE
[SPARK-33541][SQL] Group exception messages in catalyst/expressions

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR group exception messages in `/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions`.

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

**Comment by beliefer** [comment]
This PR is used to fix conflict for https://github.com/apache/spark/pull/31052

**Comment by beliefer** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala [hunk 1]
```diff
@@ -633,7 +633,7 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
         if (longValue == longValue.toByte) {
           longValue.toByte
         } else {
-          throw new ArithmeticException(s"Casting $t to byte causes overflow")
+          throw QueryExecutionErrors.castingCauseOverflowError(t, Byte.getClass.getName)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala:636]: we also need to change the codegen path.

**Review by cloud-fan — APPROVED** [review]
LGTM except one comment

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala:636]: OK

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala:636]: OK

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@allisonwang-db @maropu Thanks for your review.
@cloud-fan Thanks for your work too.

**Pull request closed without merging**